db/music.dbml

```diff
@@ -0,0 +1,299 @@
+Table submission {
+  submission_id uuid [pk, default: `gen_random_uuid()`]
+  participant_id int [not null, ref: > participant.participant_id]
+  bracket_id int [not null, ref: > bracket.bracket_id]
+  song_id int [not null, ref: > song.song_id]
+  order int [not null]
+  date_submitted timestamp [not null, default:`now()`]
+  active bool_int [not null]
+  is_dupe bool_int [not null]
+  title text [not null]
+  version text [null]
+  artist text [null]
+  collaborators text [null]
+  composer text [null]
+  link text [not null]
+  genre text [null]
+  emoji text [null]
+  comments text [null]
+
+  indexes {
+    (participant_id, bracket_id, song_id) [unique]
+  }
+}
+
+Table submission_history {
+  submission_history_id uuid [pk, default: `gen_random_uuid()`]
+  date_update timestamp [not null]
+  note_update text [not null, Note: 'dupe ineligible withdrawn "play-in loser"' ]
+  submission_id int [not null, ref: > submission.submission_id]
+  participant_id int [not null, ref: > participant.participant_id]
+  bracket_id int [not null, ref: > bracket.bracket_id]
+  song_id int [not null, ref: > song.song_id]
+  order int [not null]
+  date_submitted timestamp [not null, default:`now()`]
+  active bool_int [not null]
+  is_dupe bool_int [not null]
+  title text [not null]
+  version text [null]
+  artist text [null]
+  collaborators text [null]
+  composer text [null]
+  link text [not null]
+  genre text [null]
+  emoji text [null]
+  comments text [null]
+
+  indexes {
+    submission_id
+    participant_id
+    bracket_id
+    song_id
+  }
+
+  Note: 'composite unique index on submission_id, date_function (https://stackoverflow.com/a/26304364/2506158)'
+}
+
+Table link_relation {
+  link_realtion_id uuid [pk, default: `gen_random_uuid()`]
+  relation_id int [not null]
+  link_id int [not null, ref: > link.link_id]
+  type_link text [not null, Note: 'submission song seed' ]
+
+  indexes {
+    relation_id
+  }
+}
+
+Table link {
+  link_id uuid [pk, default: `gen_random_uuid()`]
+  url text [not null]
+  region text[] [null, Note: 'US Canada Japan Australia Europe Asia "North America" "South America" Africa Worldwide' ]
```

Comment:
This is nice and aspirational, but is overly simplistic so we definitely don't want to enforce this anywhere. YouTube videos can be turned on or off per country, and there are some that are blocked in just a single country, e.g. "everywhere but US."

Reply:
Yeah, it doesn't need to be enforced. It might be nice to have for people to quickly pick the right link, though.